images/controller/internal/controllers/rvr_status_config_node_id/reconciler_test.go

```diff
@@ -0,0 +1,767 @@
+	})
+
+	When("Get fails with non-NotFound error", func() {
+		internalServerError := errors.New("internal server error")
+		BeforeEach(func() {
+			clientBuilder = clientBuilder.WithInterceptorFuncs(InterceptGet(func(_ *v1alpha3.ReplicatedVolume) error {
+				return internalServerError
+			}))
+		})
+
+		It("should fail if getting ReplicatedVolume failed with non-NotFound error", func(ctx SpecContext) {
+			Expect(rec.Reconcile(ctx, reconcile.Request{
+				NamespacedName: types.NamespacedName{Name: "test-rv"},
+			})).Error().To(MatchError(internalServerError), "should return error when Get fails")
+		})
+	})
+
+	When("RV with RVR created", func() {
+		var (
+			rv       *v1alpha3.ReplicatedVolume
+			rvr      *v1alpha3.ReplicatedVolumeReplica
+			otherRV  *v1alpha3.ReplicatedVolume
+			otherRVR *v1alpha3.ReplicatedVolumeReplica
+		)
+
+		BeforeEach(func() {
+			rv = &v1alpha3.ReplicatedVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "volume-1",
+				},
+				Spec: v1alpha3.ReplicatedVolumeSpec{
+					Size:                       resource.MustParse("1Gi"),
+					ReplicatedStorageClassName: "test-storage-class",
+				},
+			}
+			rvr = &v1alpha3.ReplicatedVolumeReplica{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "rvr-1",
+				},
+				Spec: v1alpha3.ReplicatedVolumeReplicaSpec{
+					ReplicatedVolumeName: "volume-1",
+					NodeName:             "node-1",
+					Type:                 "Diskful",
+				},
+			}
+			Expect(controllerutil.SetControllerReference(rv, rvr, scheme)).To(Succeed())
+
+			otherRV = &v1alpha3.ReplicatedVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "volume-2",
+				},
+				Spec: v1alpha3.ReplicatedVolumeSpec{
+					Size:                       resource.MustParse("1Gi"),
+					ReplicatedStorageClassName: "test-storage-class",
+				},
+			}
+			otherRVR = &v1alpha3.ReplicatedVolumeReplica{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "rvr-vol2-1",
+				},
+				Spec: v1alpha3.ReplicatedVolumeReplicaSpec{
+					ReplicatedVolumeName: "volume-2",
+					NodeName:             "node-3",
+					Type:                 "Diskful",
+				},
+			}
+			Expect(controllerutil.SetControllerReference(otherRV, otherRVR, scheme)).To(Succeed())
+		})
+
+		JustBeforeEach(func(ctx SpecContext) {
+			if rv != nil {
+				Expect(cl.Create(ctx, rv)).To(Succeed(), "should create base RV")
+			}
+			if rvr != nil {
+				Expect(cl.Create(ctx, rvr)).To(Succeed(), "should create base RVR")
+			}
+		})
+
+		DescribeTableSubtree("when rvr has",
+			Entry("nil Status", func() { rvr.Status = nil }),
+			Entry("nil Status.DRBD", func() { rvr.Status = &v1alpha3.ReplicatedVolumeReplicaStatus{DRBD: nil} }),
+			Entry("nil Status.DRBD.Config", func() { rvr.Status = &v1alpha3.ReplicatedVolumeReplicaStatus{DRBD: &v1alpha3.DRBD{Config: nil}} }),
+			Entry("nil Status.DRBD.Config.NodeId", func() {
+				rvr.Status = &v1alpha3.ReplicatedVolumeReplicaStatus{
+					DRBD: &v1alpha3.DRBD{
+						Config: &v1alpha3.DRBDConfig{NodeId: nil},
+					},
+				}
+			}),
```

Comment:
Можно заранее гарантировать, что они не nil и тогда тут будет проще

```suggestion
			Entry("nil Status.DRBD", func() { rvr.Status.DRBD = nil }),
			Entry("nil Status.DRBD.Config", func() { rvr.Status.DRBD.Config = nil }),
			Entry("nil Status.DRBD.Config.NodeId", func() { Status.DRBD.Config.NodeId = nil }),
```